spec.bs

```diff
@@ -2800,6 +2803,143 @@ This specification defines two [=policy-controlled features=] identified by the
+  [=relevant global object=]'s [=associated Document's=] [=node navigable's=]
+  [=traversable navigable's=] [=captured ad auction headers=], |response|'s
+  [=response/header list=], and |request|'s [=request/URL=]'s [=url/origin=].
+
+<div algorithm>
+  To <dfn>update captured headers</dfn> with a [=captured ad auction headers=] |storedHeaders|,
+  [=header list=] |responseHeaders|, and [=origin=] |requestOrigin|:
+  1. Let |adAuctionSignals| be the result of [=header list/get=] with "`Ad-Auction-Signals`",
+    |responseHeaders|.
+  1. If |adAuctionSignals| is null, return.
+  1. [=header list/delete=] "`Ad-Auction-Signals`" from |responseHeaders|.
+
+    NOTE: This step prevents the header value from being used outside the intended auctions.
+  1. Let |parsedSignals| be the result of [=parsing JSON bytes to an Infra value=], given
+    |adAuctionSignals|.
+  1. If |parsedSignals| is failure, return.
+  1. If |parsedSignals| is not a [=list=], return.
+  1. [=list/For each=] |signal| in |parsedSignals|:
+    1. If |signal| is not an [=ordered map=], [=iteration/continue=].
+    1. If |signal|["`adSlot`"] doesn't exist, [=iteration/continue=].
+    1. Create a new [=direct from seller signals key=] |signalsKey|, with its
+      [=direct from seller signals key/owner=] set to |requestOrigin| and its
+      [=direct from seller signals key/ad slot=] set to |signal|["`adSlot`"].
+    1. Create a new [=direct from seller signals=] |processedSignals|.
+    1. [=map/Remove=] |signal|["`adSlot`"].
+    1. [=map/For each=] |key| → |value| of |signal|:
+      1. Switch on |key|:
+        <dl class=switch>
+        <dt>"`sellerSignals`"
+        <dd>
+        1. Set |processedSignals|'s [=direct from seller signals/seller signals=] to the result
+          of [=serializing an Infra value to a JSON string=], given |value|. If this
+          [=exception/throws=], [=iteration/continue=].
+
+        <dt>"`auctionSignals`"
+        <dd>
+        1. Set |processedSignals|'s [=direct from seller signals/auction signals=] to the result
+          of [=serializing an Infra value to a JSON string=], given |value|. If this
+          [=exception/throws=], [=iteration/continue=].
+        
+        <dt>"`perBuyerSignals`"
+        <dd>
+        1. If |value| is not an [=ordered map=], [=iteration/continue=].
+        1. For each |buyer| → |buyerSignals| of |value|:
+          1. Let |buyerOrigin| be the result of [=parsing an origin=] on |buyer|. If this
+            [=exception/throws=], [=iteration/continue=].
+          1. Let |buyerSignalsString| be the result of
+            [=serializing an Infra value to a JSON string=], given |buyerSignals|. If this
+            [=exception/throws=], [=iteration/continue=].
+          1. Set |processedSignals|'s [=direct from seller signals/per buyer signals=][|buyerOrigin|]
+            to |buyerSignalsString|.
+        
+        </dl>
+
+    1. Set |storedHeaders|[|signalsKey|] to |processedSignals|.
+</div>
+
+</div>

```

Comment:
So this is closing a div algorithm, inside which is a nested div algorithm right? That shouldn't be the structure of this. Every "Modify X to be [...]" should be in its own div algorithm. So I expect a bunch of these to be siblings of each other.

Reply:
Done